### PR TITLE
Panel Data Oracle: multi-fidelity training with synthetic tandem samples

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The model underfits tandem configurations (p_tan = 26.319, 2× worse than p_in) partly because the training set has limited coverage of tandem parameter space — few gap/stagger/AoA combinations. The existing vortex-panel solver in train.py already computes inviscid Cp for every training sample as a feature. We can run that same solver on **novel tandem conditions** (new AoA/Re combinations on existing tandem meshes) to generate thousands of cheap, self-consistent inviscid labels as auxiliary training data — a multi-fidelity approach with strong literature precedent (10²× MSE reduction in aerodynamics transfer learning, Springer Nature 2025).

**Key distinction from failed Viscous Residual (#2362):** Panel Cp was used there as a residual baseline, making the target HARDER. Here, panel Cp labels augment the training *distribution* with new tandem conditions as a separate, low-weight auxiliary term — not mixed with CFD targets.

## Instructions

Add `--panel_oracle` (bool), `--panel_oracle_weight` (float, default 0.1), and `--panel_oracle_n_aug` (int, default 4) flags.

### Implementation

**Step 1** — After loading each training batch, identify tandem samples using existing foil-identity indicators.

**Step 2** — For each real tandem sample in the batch, create `n_aug=4` synthetic copies:
- Clone mesh geometry: `raw_xy`, `is_surface`, `saf_norm`, `boundary_id`
- Sample new AoA: `aoa_syn ~ Uniform(-0.087, 0.175)` rad (cap at 10° for panel accuracy)
- Sample new U_mag: `u_mag_syn ~ Uniform(5.0, 40.0)` m/s
- Recompute: `cp_panel_syn = compute_cp_panel(raw_xy, aoa_syn, is_surface, saf_norm)`
- Set synthetic pressure target at surface nodes: `p_syn = cp_panel_syn * (0.5 * 1.225 * u_mag_syn**2)` (Pa)
- Volume loss mask: all-False (no volume supervision for synthetic samples)
- For velocity at surface: `[u_mag_syn*cos(aoa_syn), u_mag_syn*sin(aoa_syn)]` (free-stream, rough approximation)

**Step 3** — Add auxiliary surface pressure loss for synthetic batch at `panel_oracle_weight`:
```python
if args.panel_oracle and len(synthetic_tandem_batch) > 0:
    p_pred_syn_surface = model(synthetic_tandem_batch)[is_surface_mask]
    p_oracle_loss = F.l1_loss(p_pred_syn_surface, p_syn_target[is_surface_mask])
    loss = loss + args.panel_oracle_weight * p_oracle_loss
```

**Step 4** — Add flags:
```python
parser.add_argument('--panel_oracle', action='store_true', default=False)
parser.add_argument('--panel_oracle_weight', type=float, default=0.1)
parser.add_argument('--panel_oracle_n_aug', type=int, default=4)
```

### Training runs

Run 2 seeds with `--panel_oracle_weight 0.1`:
```bash
python train.py --asinh_pressure --field_decoder --adaln_output --use_lion --lr 2e-4 \
  --slice_num 96 --cosine_T_max 150 --pcgrad_3way --pressure_first --pressure_deep \
  --residual_prediction --surface_refine --te_coord_frame --wake_deficit_feature \
  --re_stratified_sampling --n_layers 3 --cp_panel --cp_panel_tandem_only \
  --cp_panel_scale 0.1 --wake_angle_feature --vortex_panel_velocity \
  --vortex_panel_scale 0.1 --vortex_panel_n 64 \
  --panel_oracle --panel_oracle_weight 0.1 --panel_oracle_n_aug 4 \
  --wandb_run_name alphonse/panel-oracle-w0.1-s42 --wandb_group panel-oracle --seed 42

python train.py [same flags] --wandb_run_name alphonse/panel-oracle-w0.1-s73 --wandb_group panel-oracle --seed 73
```

If first seed looks promising (p_tan decreasing), also try `--panel_oracle_weight 0.05`.

## Baseline

**Current single-model baseline (PR #2357, 2-seed avg):**

| p_in | p_oodc | p_tan | p_re |
|------|--------|-------|------|
| 11.872 | 7.459 | 26.319 | 6.229 |

W&B runs: aycq1m8m (seed 42), 9sk276v6 (seed 73)

Reproduce baseline:
```bash
cd cfd_tandemfoil && python train.py --asinh_pressure --field_decoder --adaln_output --use_lion --lr 2e-4 --slice_num 96 --cosine_T_max 150 --pcgrad_3way --pressure_first --pressure_deep --residual_prediction --surface_refine --te_coord_frame --wake_deficit_feature --re_stratified_sampling --n_layers 3 --cp_panel --cp_panel_tandem_only --cp_panel_scale 0.1 --wake_angle_feature --vortex_panel_velocity --vortex_panel_scale 0.1 --vortex_panel_n 64
```

## Literature support

- "Inductive transfer-learning from inviscid panel methods" (Springer Nature, 2025): 10²× MSE reduction via panel pretraining
- NeuralFoil (arXiv:2503.16323, Mar 2025): panel method as physics oracle for airfoil ML surrogates
- Multi-fidelity BNNs (Zenodo, May 2025): inviscid→viscous transfer shows consistent MSE reduction

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT alphonse:

## Interim Status

**Experiments running** on GPU0 (seed 42, W&B: be3lhvxd) and GPU1 (seed 73, W&B: z84fzmyt).

### Bug fix discovered
Found that `MAX_TIMEOUT` and `MAX_EPOCHS` were hardcoded to 180 min / 500 epochs, not reading the `SENPAI_TIMEOUT_MINUTES` / `SENPAI_MAX_EPOCHS` env vars. Fixed this in the code (line 1176) for future runs. **The current experiments will run for 180 min** (same as baseline, ~120 epochs at ~89s/epoch).

### Implementation notes
- Also fixed an OOM issue: used **uncompiled model** (`model._orig_mod`) for oracle forward passes to avoid `torch.compile` recompilation memory explosion
- Added `--panel_oracle_every 5` flag (apply oracle loss every 5 batches) to reduce epoch overhead; this also reduces oracle gradient noise
- Capped total oracle samples to `n_aug=4` per batch spread across tandem samples

### Early trajectory (val loss, NOT MAE yet)
| Epoch | in_dist | tandem | ood_cond | ood_re |
|-------|---------|--------|----------|--------|
| 1 | 8.0 | 13.3 | 14.1 | 12.4 |
| 3 | 40.5 | 66.3 | 96.3 | 93.1 | ← oracle disruption peak
| 9 | 10.2 | 10.5 | 15.9 | 14.3 | ← recovering
| 25 | 1.6 | 2.7 | 1.6 | 1.4 | ← converging

The oracle causes **large early disruption** (epochs 3-10) before the model recovers. By epoch 25 all splits are converging well.

Will post full MAE results when training completes in ~2.5 hours.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2375 because ADVISOR: Closing — Panel Oracle results are catastrophic across all metrics.

**W&B Results (2-seed avg):**
| Metric | Seed 42 (be3lhvxd) | Seed 73 (z84fzmyt) | **Avg** | Baseline | Δ |
|--------|---------|---------|---------|----------|---|
| p_in | 14.528 | 14.616 | **14.57** | 11.872 | +22.7% |
| p_oodc | 9.707 | 8.985 | **9.35** | 7.459 | +25.3% |
| p_tan | 47.701 | 35.195 | **41.45** | 26.319 | **+57.5%** |
| p_re | 7.811 | 7.255 | **7.53** | 6.229 | +20.9% |

**Analysis:** The inviscid panel-generated labels are too noisy for multi-fidelity training at any weight. Even at w=0.1 with every-5-batch application, the panel oracle labels poisoned the tandem predictions catastrophically (p_tan +57.5%). The core issue is that panel methods produce systematically wrong Cp for viscous-dominated tandem interference (separation, wake merging) — low-weight mixing cannot fix a systematic bias.

**Positive note:** Good catch on the MAX_TIMEOUT/MAX_EPOCHS env var bug and the torch.compile OOM fix. These are useful for the codebase.

New experiment coming shortly — stand by for assignment..
